compiler/ballerina-lang/src/main/java/io/ballerina/projects/JBallerinaBackend.java

```diff
@@ -572,14 +575,30 @@ private Path emitGraalExecutable(Path executableFilePath) {
                     "--no-fallback"
             };
         } else {
+            String additionalNativeArgs = "";
+            Path nativeConfigPath = packageContext.project().targetDir().resolve("cache");
             nativeImageName = project.currentPackage().packageName().toString();
-            command = new String[]{
-                    nativeImageCommand,
-                    "-jar",
+            Object graalvm = this.packageContext.packageManifest().getValue("graalvm");
+            if (graalvm != null) {
+                additionalNativeArgs = (String) ((Map) graalvm).get("additionalOptions");
+            }
+            List<String> nativeArgs = new ArrayList<>(Arrays.asList("-jar",
                     executableFilePath.toString(),
                     "-H:Name=" + nativeImageName,
                     "-H:Path=" + executableFilePath.getParent(),
-                    "--no-fallback"
+                    "--no-fallback", additionalNativeArgs));
+
+            try (FileWriter nativeArgumentWriter = new FileWriter(nativeConfigPath.resolve("native-image-args.txt")
```

Comment:
Why do we need to write it to a file? Can;t we just append the string to the command?

Reply:
When we pass too much of arguments through processbuidler, it fails on windows because of command length

Reply:
we did the same in the test side.